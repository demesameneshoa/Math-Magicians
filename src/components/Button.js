import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ name, operation }) => {
  const className = operation ? 'button operation-button' : 'button';
  return (
    <button type="button" className={className}>
      {name}
    </button>
  );
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  operation: PropTypes.bool,
};

Button.defaultProps = {
  operation: false,
};
export default Button;
