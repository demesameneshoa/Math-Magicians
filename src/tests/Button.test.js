import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Button from '../components/Button';

describe('Button component unit test', () => {
  test('Button Component Renders Number Buttons Properly ', () => {
    const tree = renderer.create(<Button name="5" clickHandler={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Button Component Renders Operation Buttons Properly', () => {
    const tree = renderer.create(<Button name="+" operation clickHandler={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
