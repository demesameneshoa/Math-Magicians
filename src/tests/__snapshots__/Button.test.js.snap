// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component unit test Button Component Renders Number Buttons Properly  1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  5
</button>
`;

exports[`Button component unit test Button Component Renders Operation Buttons Properly 1`] = `
<button
  className="button operation-button"
  onClick={[Function]}
  type="button"
>
  +
</button>
`;
